src/components/Exchange/SwapBox.js

```diff
@@ -176,8 +175,8 @@ export default function SwapBox(props) {
   const [isSubmitting, setIsSubmitting] = useState(false);
   const [modalError, setModalError] = useState(false);
   const [isHigherSlippageAllowed, setIsHigherSlippageAllowed] = useState(false);
-  const { userReferralCode } = useUserReferralCode(library, chainId, account);
-  const [userReferralCodeInLocalStorage] = useLocalStorage(REFERRAL_CODE_KEY);
+  const { isCodeAttached } = useUserReferralCode(library, chainId, account);
+  const userReferralCodeInLocalStorage = window.localStorage.getItem(REFERRAL_CODE_KEY);
```

Comment:
now `getReferralCodeTakenStatus` already returns code from localstorage, i think better to use value from there

Reply:
You mean `useUserReferralCode` instead of getReferralCodeTakenStatus I think. I will update accordingly